### PR TITLE
Frama-c 25 beta

### PR DESCRIPTION
**Pull request by bobot**

### CODE REVIEW

File: packages/frama-c/frama-c.25.0~beta/opam [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  "ocaml" { >= "4.08.1" }
+  "ocamlfind" # needed beyond build stage, used by -load-module
+  "ocamlgraph" { >= "1.8.8" }
+  "why3" { >= "1.5.0" }
+  "yojson" {>= "1.6.0"}
+  "zarith" {>= "1.5"}
+  "ppx_deriving"
+  "ppx_import"
+]
+
```
Comment (mseri) [inline packages/frama-c/frama-c.25.0~beta/opam:127]:
```suggestion

conflicts: [ 
  "result" {< "1.5"}
  "ppxlib" {< "0.9.0"}
]
```
Due to
```
# File "src/plugins/e-acsl/src/libraries/error.mli", line 26, characters 29-37:
# 26 |   type 'a result = ('a, exn) Result.t
#                                   ^^^^^^^^
# Error: Unbound type constructor Result.t
```
and
```
# Fatal error: exception File "src/plugins/value/domains/multidim/abstract_offset.ml", line 58, characters 6-10: let operators are not supported before OCaml 4.08
# File "src/plugins/value/domains/multidim/abstract_offset.ml", line 1:
# Error: Error while running external preprocessor
# Command line: /home/opam/.opam/4.08/lib/ppx_import/./ppx.exe --as-ppx '/opam-tmp/camlppx2144c3' '/opam-tmp/camlppx18606f'
```

File: packages/frama-c/frama-c.25.0~beta/opam [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+conflicts: [ 
+  "result" {< "1.5"}
+  "ppxlib" {< "0.9.0"}
+]
```
Comment (mseri) [inline packages/frama-c/frama-c.25.0~beta/opam:131]:
```suggestion
  "cairo2" { < "0.6.2" }
]
```
Due to
```
 /usr/bin/ld: /home/opam/.opam/4.14/lib/cairo2/libcairo_stubs.a(cairo_stubs.o):/home/opam/.opam/4.14/.opam-switch/build/cairo2.0.6/_build/default/src/cairo_ocaml.h:53: multiple definition of `caml_cairo_font_type'; /home/opam/.opam/4.14/lib/lablgtk3/liblablgtk3_stubs.a(cairo_pango_stubs.o):/home/opam/.opam/4.14/lib/cairo2/cairo_ocaml.h:53: first defined here
# /usr/bin/ld: /home/opam/.opam/4.14/lib/cairo2/libcairo_stubs.a(cairo_stubs.o):/home/opam/.opam/4.14/.opam-switch/build/cairo2.0.6/_build/default/src/cairo_ocaml.h:28: multiple definition of `caml_cairo_ops'; /home/opam/.opam/4.14/lib/lablgtk3/liblablgtk3_stubs.a(cairo_pango_stubs.o):/home/opam/.opam/4.14/lib/cairo2/cairo_ocaml.h:28: first defined here
# /usr/bin/ld: /home/opam/.opam/4.14/lib/cairo2/libcairo_stubs.a(cairo_stubs.o):/home/opam/.opam/4.14/.opam-switch/build/cairo2.0.6/_build/default/src/cairo_ocaml.h:70: multiple definition of `caml_surface_ops'; /home/opam/.opam/4.14/lib/lablgtk3/liblablgtk3_stubs.a(cairo_pango_stubs.o):/home/opam/.opam/4.14/lib/cairo2/cairo_ocaml.h:70: first defined here
# /usr/bin/ld: /home/opam/.opam/4.14/lib/cairo2/libcairo_stubs.a(cairo_stubs.o):/home/opam/.opam/4.14/.opam-switch/build/cairo2.0.6/_build/default/src/cairo_ocaml.h:36: multiple definition of `caml_pattern_ops'; /home/opam/.opam/4.14/lib/lablgtk3/liblablgtk3_stubs.a(cairo_pango_stubs.o):/home/opam/.opam/4.14/lib/cairo2/cairo_ocaml.h:36: first defined here
# /usr/bin/ld: /home/opam/.opam/4.14/lib/cairo2/libcairo_stubs.a(cairo_stubs.o):/home/opam/.opam/4.14/.opam-switch/build/cairo2.0.6/_build/default/src/cairo_ocaml.h:96: multiple definition of `caml_path_ops'; /home/opam/.opam/4.14/lib/lablgtk3/liblablgtk3_stubs.a(cairo_pango_stubs.o):/home/opam/.opam/4.14/lib/cairo2/cairo_ocaml.h:96: first defined here
# /usr/bin/ld: /home/opam/.opam/4.14/lib/cairo2/libcairo_stubs.a(cairo_stubs.o):/home/opam/.opam/4.14/.opam-switch/build/cairo2.0.6/_build/default/src/cairo_ocaml.h:48: multiple definition of `caml_font_options_ops'; /home/opam/.opam/4.14/lib/lablgtk3/liblablgtk3_stubs.a(cairo_pango_stubs.o):/home/opam/.opam/4.14/lib/cairo2/cairo_ocaml.h:48: first defined here
# /usr/bin/ld: /home/opam/.opam/4.14/lib/cairo2/libcairo_stubs.a(cairo_stubs.o):/home/opam/.opam/4.14/.opam-switch/build/cairo2.0.6/_build/default/src/cairo_ocaml.h:64: multiple definition of `caml_scaled_font_ops'; /home/opam/.opam/4.14/lib/lablgtk3/liblablgtk3_stubs.a(cairo_pango_stubs.o):/home/opam/.opam/4.14/lib/cairo2/cairo_ocaml.h:64: first defined here
# /usr/bin/ld: /home/opam/.opam/4.14/lib/cairo2/libcairo_stubs.a(cairo_stubs.o):/home/opam/.opam/4.14/.opam-switch/build/cairo2.0.6/_build/default/src/cairo_ocaml.h:154: multiple definition of `caml_cairo_ft_library_ops'; /home/opam/.opam/4.14/lib/lablgtk3/liblablgtk3_stubs.a(cairo_pango_stubs.o):/home/opam/.opam/4.14/lib/cairo2/cairo_ocaml.h:154: first defined here
# /usr/bin/ld: /home/opam/.opam/4.14/lib/cairo2/libcairo_stubs.a(cairo_stubs.o):/home/opam/.opam/4.14/.opam-switch/build/cairo2.0.6/_build/default/src/cairo_ocaml.h:157: multiple definition of `caml_cairo_ft_face_ops'; /home/opam/.opam/4.14/lib/lablgtk3/liblablgtk3_stubs.a(cairo_pango_stubs.o):/home/opam/.opam/4.14/lib/cairo2/cairo_ocaml.h:157: first defined here
# collect2: error: ld returned 1 exit status
```

**Comment by mseri** [comment]
The `dune-configurator` error is fixed by https://github.com/ocaml/opam-repository/pull/21393

File: packages/frama-c/frama-c.25.0~beta/opam [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+  "conf-graphviz" { post }
+  "conf-time" { with-test }
+  "ocaml" { >= "4.08.1" }
+  "ocamlfind" # needed beyond build stage, used by -load-module
+  "ocamlgraph" { >= "1.8.8" }
+  "why3" { >= "1.5.0" }
+  "yojson" {>= "1.6.0"}
+  "zarith" {>= "1.5"}
+  "ppx_deriving"
+  "ppx_import"
```
Comment (mseri) [inline packages/frama-c/frama-c.25.0~beta/opam:125]:
```suggestion
  "ppx_import" {>= "1.8.0"}
```
The problem was not ppxlib but the version of ppx_import that uses 4.08+ AST

File: packages/frama-c/frama-c.25.0~beta/opam [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+]
+
+conflicts: [ 
+  "result" {< "1.5"}
+  "ppxlib" {< "0.9.0"}
```
Comment (mseri) [inline packages/frama-c/frama-c.25.0~beta/opam:130]:
```suggestion
```

**Comment by mseri** [comment]
Are the revdeps failing due to breaking changes in some frama-c related parsetree?
```
# File "concurrency_model/concurrency_model.ml", line 96, characters 6-28:
# 96 |     | TComp (compinfo, _, _) -> 
#            ^^^^^^^^^^^^^^^^^^^^^^
# Error: The constructor TComp expects 2 argument(s),
#        but is applied here to 3 argument(s)
```
```
# File "meta_bindings.ml", lines 122-123, characters 18-55:
# 122 | ..................TArray (base_type, Some (Cil.integer unkloc n),
# 123 |                           {scache = Not_Computed} , [])
# Error: The constructor TArray expects 3 argument(s),
#        but is applied here to 4 argument(s)
```

**Comment by maroneze** [comment]
Yes, the revdeps failures are likely due to breaking changes. This is an extract of the beta "main changes" log:

> - significant changes in AST types that might break existing development

I'll see if we can propose patches for some of them, but from what I see, it's just a few expected syntactic changes.

**Pull request merged**